src/main/java/org/terasology/taskSystem/actions/FindNextBlockOnBoundaryNode.java

```diff
@@ -103,30 +105,24 @@ private void setTargetToNearbyBoundaryBlock(Actor oreon, TaskComponent taskCompo
 
         if (Math.round(worldPosition.x) == maxX) {
             if (Math.round(worldPosition.z) != maxZ) {
-                moveComponent.target = new Vector3f(maxX + 1, y, maxZ + 1);
+                moveComponent.target = JomlUtil.from(new Vector3f(maxX + 1, y, maxZ + 1));
```

Comment:
Can't this be replaced by `moveComponent.target.set(...)` everywhere, with a single check whether the field is null?